accounts.go

```diff
@@ -0,0 +1,385 @@
+const (
+	accountTypeBank       = "BANK"
+	accountTypeCurrent    = "CURRENT"
+	accountTypeCurrLiab   = "CURRLIAB"
+	accountTypeDepreciatn = "DEPRECIATN"
+	accountTypeDC         = "DIRECTCOSTS"
+	accountTypeEquity     = "EQUITY"
+	accountTypeExp        = "EXPENSE"
+	accountTypeFixed      = "FIXED"
+	accountTypeInventory  = "INVENTORY"
+	accountTypeLiab       = "LIABILITY"
+	accountTypeNonCurrent = "NONCURRENT"
+	accountTypeOther      = "OTHERINCOME"
+	accountTypeOverhead   = "OVERHEADS"
+	accountTypePrepay     = "PREPAYMENT"
+	accountTypeRevenue    = "REVENUE"
+	accountTypeSale       = "SALES"
+	accountTypeTermLiab   = "TERMLIAB"
+	accountTypePAYG       = "PAYGLIABILITY"
+	accountTypeSAExp      = "SUPERANNUATIONEXPENSE"
+	accountTypeSALiab     = "SUPERANNUATIONLIABILITY"
+	accountTypeWageExp    = "WAGESEXPENSE"
+	accountTypeWageLiab   = "WAGESPAYABLELIABILITY"
+)
+
+// Xero Account types
+var (
+	AccountTypeBank       = AccountType{accountTypeBank}
+	AccountTypeCurrent    = AccountType{accountTypeCurrent}
+	AccountTypeCurrLiab   = AccountType{accountTypeCurrLiab}
+	AccountTypeDepreciatn = AccountType{accountTypeDepreciatn}
+	AccountTypeDC         = AccountType{accountTypeDC}
+	AccountTypeEquity     = AccountType{accountTypeEquity}
+	AccountTypeExp        = AccountType{accountTypeExp}
+	AccountTypeFixed      = AccountType{accountTypeFixed}
+	AccountTypeInventory  = AccountType{accountTypeInventory}
+	AccountTypeLiab       = AccountType{accountTypeLiab}
+	AccountTypeNonCurrent = AccountType{accountTypeNonCurrent}
+	AccountTypeOther      = AccountType{accountTypeOther}
+	AccountTypeOverhead   = AccountType{accountTypeOverhead}
+	AccountTypePrepay     = AccountType{accountTypePrepay}
+	AccountTypeRevenue    = AccountType{accountTypeRevenue}
+	AccountTypeSale       = AccountType{accountTypeSale}
+	AccountTypeTermLiab   = AccountType{accountTypeTermLiab}
+	AccountTypePAYG       = AccountType{accountTypePAYG}
+	AccountTypeSAExp      = AccountType{accountTypeSAExp}
+	AccountTypeSALiab     = AccountType{accountTypeSALiab}
+	AccountTypeWageExp    = AccountType{accountTypeWageExp}
+	AccountTypeWageLiab   = AccountType{accountTypeWageLiab}
+)
+
+// The AccountType type defines the specific account types within Xero:
+type AccountType struct {
+	value string
+}
+
+// String implements the Stringer interface returning the string representation
+// of the AccountType
+func (a AccountType) String() string {
+	return a.value
+}
+
+// MarshalXML marshals a AccountType into valid XML for Xero
+func (a *AccountType) MarshalXML(encoder *xml.Encoder, start xml.StartElement) error {
+	return encoder.EncodeElement(a.value, start)
+}
+
+// unmarshalXML handles converting raw Xero AccountType XML data into valid AccountType
+func (a *AccountType) unmarshalXML(decoder elementDecoder, start xml.StartElement) error {
+	var value string
+	if err := decoder.DecodeElement(&value, &start); err != nil {
+		return err
+	}
+	switch value {
```

Comment:
Yeh I wasn't happy with how massive this switch statement was getting, good idea with a for loop that should reduce the size, will update :+1: